src/mscorlib/src/System/Guid.cs

```diff
@@ -365,214 +365,287 @@ public static bool TryParseExact(String input, String format, out Guid result)
             }
         }
 
-
-        private static bool TryParseGuid(String g, GuidStyles flags, ref GuidResult result)
+        private static bool TryParseGuid(string g, GuidStyles flags, ref GuidResult result)
         {
-            if (g == null) {
+            if (g == null)
+            {
                 result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
                 return false;
             }
-            String guidString = g.Trim();  //Remove Whitespace
+            
+            // Consume leading and trailing whitespace
+            int start = 0;
+            while (start < g.Length && char.IsWhiteSpace(g[start]))
+            {
+                start++;
+            }
 
-            if (guidString.Length == 0) {
+            int end = g.Length;
+            while (end > start && char.IsWhiteSpace(g[end - 1]))
+            {
+                end--;
+            }
+
+            int length = end - start;
+            if (length == 0)
+            {
+                // Empty string (or only whitespace)
                 result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
                 return false;
             }
 
-            // Check for dashes
-            bool dashesExistInString = (guidString.IndexOf('-', 0) >= 0);
-
-            if (dashesExistInString) {
-                if ((flags & (GuidStyles.AllowDashes | GuidStyles.RequireDashes)) == 0) {
-                    // dashes are not allowed
-                    result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
-                    return false;
-                }
-            }
-            else {
-                if ((flags & GuidStyles.RequireDashes) != 0) {
-                    // dashes are required
-                    result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
-                    return false;
+            bool hasDashesInString = false;
+            bool hasBracesInString = false;
+            bool hasParenthesesInString = false;
+            for (int i = start; i < end; i++)
+            {
+                char c = g[i];
+                if (c == '-')
+                {
+                    if ((flags & (GuidStyles.AllowDashes | GuidStyles.RequireDashes)) == 0)
```

Comment:
NIT: If I was making the changes here, I'd keep the structure we had before.  I'd pull your loop up that sets the condition variables (so we only walk the string once), but keep the code that used to be on L385, L403 and L421 the same.

Again, this is to just reduce noise in the diff.  I understand how your way allows us to early out in failure cases, but I'd trade off the small additional string walking we have to do in failure cases in order to have a smaller diff.